docs/eks-v3-migration.md

```diff
@@ -0,0 +1,148 @@
+# EKS v3 Migration Guide
+
+AWS recently announced the deprecation of two features used by default in Pulumi EKS: the aws-auth ConfigMap and the AL2 operating system. Pulumi EKS v3 addresses these deprecations, enhances the maintainability of the provider, and aligns it with EKS best practices.
+
+## New Features
+
+- **Custom AMI, GPU Node, and User Data Support for `ManagedNodeGroup`**  
+  `ManagedNodeGroup` now allows greater customization of node groups by supporting custom Amazon Machine Images (AMIs). This feature is particularly useful for users who require specific configurations or optimizations in their node images.  
+  Additionally, you can now automatically select the right AMIs for nodes with GPUs by setting the `gpu` flag.  
+  You can now also configure custom user data, giving more flexibility to users who need to run specific scripts or commands on node startup.  
+    
+- **Default Instance Type Update: t3.medium**  
+  The default instance type for node groups has been updated from `t2.medium` to `t3.medium`. The `t3.medium` instance type provides several advantages:  
+    
+  - **Higher Availability**: `t2.medium` is an older instance type that is being phased out in AWS regions, which can cause capacity issues during scaling. The newer `t3.medium` instance type offers better availability.
+  - **Improved Cost Efficiency**: `t3.medium` is marginally cheaper than `t2.medium` ($0.0416 vs $0.0464 per hour), providing better performance at a lower cost.
+
+
+- **New `operatingSystem` Property for Node Groups**  
+  A new `operatingSystem` property has been introduced for node groups, giving users the flexibility to choose which OS their nodes should run. Supported options include Amazon Linux 2023 (AL2023) and Bottlerocket, allowing you to select the OS that best fits your workloads and compliance needs.  
+    
+  - **`RECOMMENDED` Default Setting**: By default, the `RECOMMENDED` option will be used, which ensures your nodes are always running the OS recommended by AWS for optimal security and performance. This default setting will be kept in sync with AWS’s updates.
+
+
+- **EKS Cluster Addon Management: `vpc-cni` and `coredns`**  
```

Comment:
Should we also mention `kube-proxy`?

Also, I think we need to add something about cases where users are already managing these with addons. They will have to explicitly disable the creation of these in the cluster.